compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -281,10 +313,32 @@ public void visit(BLangFunction funcNode) {
             addToFunctionMap(funcNode, funcEnv, position, receiver.symbol, receiver.getBType());
         }
 
+        if (funcNode.flagSet.contains(Flag.LAMBDA)) {
```

Comment:
Why do we have to handle the closure and non-closure cases separately?

Reply:
Because closures for parameters in normal function should be created first. for lambda functions, we can create closures when visiting the expr. 
example
```
function foo() {
    value();
}

function value(int a = 10) {

}
```
Here closures related to the parameters of function `value` should create before visiting the function body of `foo`

Reply:
Removed with new changes.